src/cascadia/TerminalCore/Terminal.hpp

```diff
@@ -276,6 +277,7 @@ class Microsoft::Terminal::Core::Terminal final :
     size_t _hyperlinkPatternId;
 
     std::wstring _workingDirectory;
+    std::optional<FontInfo> _fontInfo;

```

Comment:
like: TSFInputControl had the same problem -- had to get the font -- and I am concerned that now there are two ways to get the font (new one and old one)

Reply:
So this is the part that sucks. `IUiaData` is down in `TerminalCore`. But all of the font info (`_desiredFont` and `_actualFont`) is actually up in `ControlCore`. Because of this, we need to have `ControlCore` tell `TerminalCore` "hey, the font change to X" [here](https://github.com/microsoft/terminal/pull/10336#discussion_r660842660).

`TSFInputControl` looks like it works directly with the `TerminalControl` (now `ControlCore` since the split) to get what it needs.